templates/zimrc

```diff
@@ -12,7 +12,7 @@
 # The second line of modules may depend on options set by modules in the first
 # line. These dependencies are noted on the respective module's README.md.
 zmodules=(directory environment git git-info history input utility custom \
-          syntax-highlighting history-substring-search prompt completion)
+          syntax-highlighting prompt completion history-substring-search)
```

Comment:
Why the order change? `history-substring-search` preferably should not be before `input`, and `completion` should be the last.

Reply:
It was suggested as part of the fix. It seems to work fine on my machine with it in the usual place, though.

Reply:
We need to figure out some way to enforce module loading order to some degree, or alternatively introduce some "bootstrap" and "post-init" conditionals prior to module loading to handle some of these cases (editor mode, completions, etc).

Reply:
Sure, a generic solution would be much better, but this is a start.

Reply:
Agreed. I think it's possible to make the modules independent of each other, so they can be loaded in any order. The changes proposed here adhere to this idea.

Side comment: Maybe we should move the `compinit` code to `init.zsh` also, to guarantee the same for the `completion` module.